napari/_qt/qt_main_window.py

```diff
@@ -769,20 +786,36 @@ def _add_menus(self):
         self._main_menu_shortcut.activated.connect(
             self._toggle_menubar_visible
         )
-
-        self.file_menu = menus.FileMenu(self)
+        # file menu
+        self.file_menu = build_qmodel_menu(
+            MenuId.MENUBAR_FILE, title=trans._('&File'), parent=self._qt_window
+        )
+        self._setup_npe1_samples_menu()
+        self.file_menu.aboutToShow.connect(
+            lambda: self._update_enabled('file_menu')
+        )
```

Comment:
> Actually - is it possible this is contributing to the docs failures?

Tried and it still fails without lambda: https://github.com/napari/napari/pull/4865/commits/9fdec0ef170c8781d0d57eb2e0ea29a9a5898f60

I could not think of a good way of doing this that would not require a new `_update_enabled` function for each menu. If you have a good solution, happy to use, otherwise I will disconnect it in teardown.

Reply:
Ah I don't think I can disconnect a lambda (I am not familiar with this though). AFAICT we can either connect to lambda or create separate update functions for each menu and connect to these. WDYT?

Reply:
> I played with this as well for a few hours but couldn't fix the tests or observe different behavior.

Thank you! It has stumped me too.

Will leave as is unless someone has a good solution, thanks!